// brawlstars-api-nodejs/lib/_classes/BattleLog.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Battle = void 0;
/**
 * Represents a battle in Brawl Stars.
 * @class
 */
class Battle {
    /**
     * The time the battle took place.
     * @type {string}
     */
    time;
    /**
     * The mode in which the battle was played.
     * @type {Mode}
     */
    mode;
    /**
     * The map on which the battle was played.
     * @type {string}
     */
    map;
    /**
     * The type of battle.
     * @type {BattleType}
     */
    type;
    /**
     * The result of the battle.
     * @type {BattleResult}
     */
    result;
    /**
     * The duration of the battle in seconds.
     * @type {number}
     */
    duration;
    /**
     * The player's rank at the end of the battle.
     * @type {number}
     */
    rank;
    /**
     * The change in the player's trophies after the battle.
     * @type {number}
     */
    trophyChange;
    /**
     * The star player of the battle.
     * @type {null | StarPlayer}
     */
    starPlayer;
    /**
     * The teams that participated in the battle.
     * @type {[APIBattlePlayer[], APIBattlePlayer[], APIBattlePlayer[]?, APIBattlePlayer[]?, APIBattlePlayer[]?, APIBattlePlayer[]?, APIBattlePlayer[]?, APIBattlePlayer[]?, APIBattlePlayer[]?, APIBattlePlayer[]?]}
     */
    teams;
    /**
     * Creates a new Battle instance.
     * @param {APIBattle} api The battle data returned by the API.
     * @constructor
     */
    constructor(api) {
        this.time = api.battleTime;
        api.event.mode ? (this.mode = api.event.mode) : (this.mode = "unknown");
        api.event.map ? (this.map = api.event.map) : (this.map = "Community map");
        this.type = api.battle.type;
        if (api.battle.result)
            this.result = api.battle.result;
        if (api.battle.duration)
            this.duration = api.battle.duration;
        if (api.battle.rank)
            this.rank = api.battle.rank;
        if (api.battle.trophyChange)
            this.trophyChange = api.battle.trophyChange;
        if (!api.battle.starPlayer) {
            this.starPlayer = null;
        }
        else {
            this.starPlayer = {
                tag: api.battle.starPlayer.tag,
                name: api.battle.starPlayer.name,
                brawler: {
                    name: api.battle.starPlayer.brawler.name,
                    power: api.battle.starPlayer.brawler.power,
                    trophies: api.battle.starPlayer.brawler.trophies,
                },
            };
        }
        this.teams = api.battle.teams;
    }
    /**
     * Returns an array of all the player tags in the battle.
     * @returns {string[]}
     */
    getTags() {
        let tags = [];
        for (let i = 0; i < this.teams.length; i++) {
            for (let j = 0; j < this.teams[i].length; j++) {
                tags.push(this.teams[i][j].tag);
            }
        }
        return tags;
    }
    /**
     * Returns an array of all the player names in the battle.
     * @returns {string[]}
     */
    getNames() {
        let names = [];
        for (let i = 0; i < this.teams.length; i++) {
            for (let j = 0; j < this.teams[i].length; j++) {
                names.push(this.teams[i][j].name);
            }
        }
        return names;
    }
}
exports.Battle = Battle;
/**
 * Represents a log of battles in Brawl Stars.
 * @class
 */
class BattleLog {
    /**
     * The history of battles.
     * @type {Battle[]}
     */
    history = [];
    /**
     * Creates a new BattleLog instance.
     * @param {APIBattle[]} api The battle log data returned by the API.
     * @constructor
     */
    constructor(api) {
        for (let i = 0; i < api.length; i++) {
            this.history.push(new Battle(api[i]));
        }
    }
    /**
     * Returns the most recent battle in the log.
     * @returns {Battle}
     */
    last() {
        return this.history[0];
    }
    /**
     * Returns the specified battle in the log.
     * @param {number} number The index of the battle to return, with the most recent battle being 1.
     * @returns {Battle}
     */
    game(number) {
        return this.history[number - 1];
    }
}
exports.default = BattleLog;
