// brawlstars-api-nodejs/lib/_classes/BrawlStarsClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const BattleLog_1 = require("./BattleLog");
const BrawlStarsApi_1 = require("./BrawlStarsApi");
const Club_1 = require("./Club");
const ClubRanking_1 = require("./ClubRanking");
const Player_1 = require("./Player");
const PlayerRanking_1 = require("./PlayerRanking");
/**
 * Class that represents a client for the Brawl Stars API
 * @class
 */
class Client {
    /**
     * The instance of the BrawlStarsApi class.
     * @type {API}
     * @private
     */
    api;
    /**
     * Creates a new instance of the Client class.
     * @param {string} token The API token.
     * @constructor
     */
    constructor(token) {
        this.api = new BrawlStarsApi_1.default(token);
    }
    /**
     * Retrieves a player's data from the Brawl Stars API.
     * @param {string} tag The player's tag, formatted as "#2R20PL0UR"
     * @returns {Promise<Player>} A promise that resolves to the player's data.
     */
    async player(tag) {
        let api = await this.api.player(tag);
        let client = new Player_1.default(api);
        return client;
    }
    /**
     * Retrieves the battle log for the player with the given tag.
     * @param {string} tag - The tag of the player to retrieve the battle log for.
     * @return {Promise<BattleLog>} A promise that resolves to an array of objects representing the player's battle log.
     */
    async battleLog(tag) {
        let api = await this.api.battleLog(tag);
        let client = new BattleLog_1.default(api);
        return client;
    }
    /**
     * Returns information about a specific club.
     * @param {string} tag - Club tag
     * @returns {Promise<IClub>} Club information
     */
    async club(tag) {
        let api = await this.api.club(tag);
        let client = new Club_1.default(api);
        return client;
    }
    /**
     * Retrieves the list of members of a club.
     * @param {string} tag - The tag of the club to retrieve information for.
     * @returns {Promise<Member[]>} An array of objects representing the members of the club.
     */
    async clubMembers(tag) {
        return (await this.club(tag)).members;
    }
    /**
     * Returns a list of all brawlers in the game.
     * @returns {Promise<APIBrawler[]>} A promise that resolves to an array of brawlers.
     */
    async brawlers() {
        return await this.api.brawlers();
    }
    /**
     * Retrieve information about a specific brawler
     * @param {number} id - The id brawler
     * @returns {Promise<APIBrawler>} A promise that resolves to an object containing information about the brawler
     */
    async brawler(id) {
        return await this.api.brawler(id);
    }
    /**
     * Returns an array of seasons for the specified country's Power Play ranking.
     * @param {Country} [counrty="global"] - counrty The country to get the Power Play ranking seasons for. Defaults to 'global'.
     * @returns {Promise<APIPowerPlayLeague[]>} An array of seasons for the specified country's Power Play ranking.
     */
    async powerPlayRankingSeasons(counrty = "global") {
        return this.api.powerPlayRankingSeasons(counrty);
    }
    /**
     * Get the trophy ranking for a specific country.
     * @param {Country} [counrty="global"] - counrty The country to get the ranking for. Defaults to "global".
     * @returns {Promise<IRankingPlayer[]>} An array of players ranked by their trophies.
     */
    async trophyRanking(counrty = "global") {
        let api = await this.api.trophyRanking(counrty);
        let client = new PlayerRanking_1.default(api);
        return client;
    }
    /**
     * Get the power play ranking for a specific country and season.
     * @param {(number | "latest")} id The ID of the season. Defaults to "latest".
     * @param {Country} [counrty="global"] - counrty The country to get the ranking for. Defaults to "global".
     * @returns {Promise<IRankingPlayer[]>} An array of players ranked by their power play points.
     */
    async powerPlayRanking(id = "latest", counrty = "global") {
        let api = await this.api.powerPlayRanking(id, counrty);
        let client = new PlayerRanking_1.default(api);
        return client;
    }
    /**
     * @param {number | string} id - The brawler ID or name.
     * @param {Country} [counrty="global"] - The country code.
     * @returns {Promise<PlayerRanking>} A promise that resolves with the player ranking object.
     */
    async brawlerRanking(id, counrty = "global") {
        let api = await this.api.brawlerRanking(id, counrty);
        let client = new PlayerRanking_1.default(api);
        return client;
    }
    /**
     * Returns the club ranking for the specified country.
     * @param {Country} [counrty="global"] The country to get the ranking for.
     * @returns {Promise<ClubRanking>} A promise that resolves to the club ranking.
     */
    async clubRanking(counrty = "global") {
        let api = await this.api.clubRanking(counrty);
        let client = new ClubRanking_1.default(api);
        return client;
    }
    /**
     * Returns the current rotation of maps in the game.
     * @returns {Promise<APIMap>} The current rotation of maps.
     */
    async rotation() {
        return this.api.rotation();
    }
}
exports.default = Client;
