// brawlstars-api-nodejs/lib/_classes/ClubRanking.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * A class for storing and manipulating a list of ranked clubs.
 * @class ClubRanking
 */
class ClubRanking {
    /**
     * The list of ranked clubs.
     * @type {RankingClub[]}
     */
    ranking = [];
    /**
     * Creates an instance of ClubRanking.
     * @param {APIRankingClub[]} api - The list of ranked clubs from the API.
     * @constructor
     */
    constructor(api) {
        for (let i = 0; i < api.length; i++) {
            this.ranking.push({
                tag: api[i].tag,
                name: api[i].name,
                trophies: api[i].trophies,
                rank: api[i].rank,
                memberCount: api[i].memberCount,
            });
        }
    }
    /**
     * Returns the club at the specified rank.
     * @param {number} position - The rank of the club.
     * @returns {(null | RankingClub)} The club at the specified rank, or null if the rank is invalid.
     */
    rank(position) {
        if (this.ranking[position - 1]) {
            return this.ranking[position - 1];
        }
        return null;
    }
    /**
     * Returns an array of club tags in the ranking.
     * @returns {string[]} An array of club tags.
     */
    getTags() {
        let tags = [];
        for (let i = 0; i < this.ranking.length; i++) {
            tags.push(this.ranking[i].tag);
        }
        return tags;
    }
    /**
     * Returns an array of club names in the ranking.
     * @returns {string[]} An array of club names.
     */
    getNames() {
        let names = [];
        for (let i = 0; i < this.ranking.length; i++) {
            names.push(this.ranking[i].name);
        }
        return names;
    }
    /**
     * Returns an array of clubs in the ranking with fewer than 30 members.
     * @returns {RankingClub[]} An array of clubs with fewer than 30 members.
     */
    notFull() {
        return this.ranking.filter((club) => club.memberCount < 30);
    }
}
exports.default = ClubRanking;
