// brawlstars-api-nodejs/lib/_classes/Club.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Represents a club in Brawl Stars.
 * @class
 */
class Club {
    /**
     * The club's tag.
     * @type {string}
     */
    tag;
    /**
     * The club's name.
     * @type {string}
     */
    name;
    /**
     * The club's description.
     * @type {string}
     */
    description;
    /**
     * The club's type.
     * @type {ClubType}
     */
    type;
    /**
     * The club's current number of trophies.
     * @type {number}
     */
    trophies;
    /**
     * The club's required number of trophies for membership.
     * @type {number}
     */
    requiredTrophies;
    /**
     * The club's members.
     * @type {Member[]}
     */
    members = [];
    /**
     * The club's total number of members.
     * @type {number}
     */
    count_members;
    /**
     * The club's status, either "full" or "not full".
     * @type {("full" | "not full")}
     */
    status;
    /**
     * Creates a new Club instance.
     * @param {APIClub} api The club data returned by the API.
     * @constructor
     */
    constructor(api) {
        this.tag = api.tag;
        this.name = api.name;
        this.description = api.description;
        this.type = api.type;
        this.trophies = api.trophies;
        this.requiredTrophies = api.requiredTrophies;
        for (let i = 0; i < api.members.length; i++) {
            this.members.push({
                tag: api.members[i].tag,
                name: api.members[i].name,
                trophies: api.members[i].trophies,
                role: api.members[i].role,
            });
        }
        this.count_members = this.members.length;
        this.count_members == 30 ? (this.status = "full") : (this.status = "not full");
    }
    /**
     * Returns an array of all the member tags in the club.
     * @returns {string[]}
     */
    getTags() {
        let tags = [];
        for (let i = 0; i < this.members.length; i++) {
            tags.push(this.members[i].tag);
        }
        return tags;
    }
    /**
     * Returns an array of all the member names in the club.
     * @returns {string[]}
     */
    getNames() {
        let names = [];
        for (let i = 0; i < this.members.length; i++) {
            names.push(this.members[i].name);
        }
        return names;
    }
    /**
     * Returns an array of all the members in the club with a specific role.
     * @param {ClubMemberRole} role The role to filter by.
     * @returns {Member[]}
     */
    getRoleMembers(role) {
        return this.members.filter((member) => member.role == role);
    }
}
exports.default = Club;
