// brawlstars-api-nodejs/lib/_classes/Player.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * A class representing a player in the game.
 * @class Player
 */
class Player {
    /**
     * The player's tag.
     * @type {string}
     */
    tag;
    /**
     * The player's name.
     * @type {string}
     */
    name;
    /**
     * The player's current number of trophies.
     * @type {number}
     */
    trophies;
    /**
     * The player's highest number of trophies achieved.
     * @type {number}
     */
    highestTrophies;
    /**
     * The player's highest number of trophies achieved.
     * @type {{trio: number, duo: number, solo: number}};
     */
    victories;
    /**
     * An object containing the player's club information.
     * @type {{tag: string, name: string}};
     */
    club;
    /**
     * An object containing information about each brawler the player has.
     * @type { Brawler[] }
     */
    brawlers = {};
    /**
     * Creates a new `Player` instance from an API response.
     * @param {APIPlayer} api The API response containing player data.
     * @constructor
     */
    constructor(api) {
        this.tag = api.tag;
        this.name = api.name;
        this.trophies = api.trophies;
        this.highestTrophies = api.highestTrophies;
        this.victories = {
            trio: api["3vs3Victories"],
            duo: api.duoVictories,
            solo: api.soloVictories,
        };
        this.club = api.club;
        for (let i = 0; i < api.brawlers.length; i++) {
            let brawler = api.brawlers[i];
            this.brawlers[brawler.name.charAt(0) + brawler.name.slice(1).toLowerCase()] = {
                rank: brawler.rank,
                trophies: brawler.trophies,
                highestTrophies: brawler.highestTrophies,
                power: brawler.power,
                countStarPowers: brawler.starPowers.length,
                countGadgets: brawler.gadgets.length,
                countGears: brawler.gears.length,
            };
        }
    }
    /**
     * Returns the total number of wins the player has across all game modes.
     * @returns {number} The total number of wins.
     */
    wins() {
        return this.victories.trio + this.victories.duo + this.victories.solo;
    }
    /**
     * Returns the player's brawler with the specified name.
     * @param {string} name The name of the brawler to retrieve.
     * @returns {(null | Brawler)} The brawler, or `null` if the brawler with the specified name does not exist.
     */
    brawler(name) {
        if (Object.keys(this.brawlers).includes(name)) {
            return this.brawlers[name];
        }
        return null;
    }
}
exports.default = Player;
