// brawlstars-api-nodejs/lib/_classes/PlayerRanking.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * A class for storing and manipulating a list of ranked players.
 * @class PlayerRanking
 */
class PlayerRanking {
    /**
     * The list of ranked players.
     * @type {RankingPlayer[]}
     */
    ranking = [];
    /**
     * Creates a new `PlayerRanking` instance from an API response.
     * @param {APIRankingPlayer[]} api The API response containing player ranking data.
     * @constructor
     */
    constructor(api) {
        for (let i = 0; i < api.length; i++) {
            this.ranking.push({
                tag: api[i].tag,
                name: api[i].name,
                trophies: api[i].trophies,
                rank: api[i].rank,
            });
            if (api[i].club) {
                this.ranking[i].club = api[i].club.name;
            }
        }
    }
    /**
     * Returns the player at the specified position in the ranking list.
     * @param {number} position The rank of the player to retrieve.
     * @returns {(null | RankingPlayer)} The player at the specified position, or `null` if the position is out of bounds.
     */
    rank(position) {
        if (this.ranking[position - 1]) {
            return this.ranking[position - 1];
        }
        return null;
    }
    /**
     * Returns a list of all player tags in the ranking list.
     * @returns {string[]} An array of player tags.
     */
    getTags() {
        let tags = [];
        for (let i = 0; i < this.ranking.length; i++) {
            tags.push(this.ranking[i].tag);
        }
        return tags;
    }
    /**
     * Returns a list of all player names in the ranking list.
     * @returns {string[]} An array of player names.
     */
    getNames() {
        let names = [];
        for (let i = 0; i < this.ranking.length; i++) {
            names.push(this.ranking[i].name);
        }
        return names;
    }
}
exports.default = PlayerRanking;
