// brawlstars-api-nodejs/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.API = exports.Client = exports.PlayerRanking = exports.Player = exports.ClubRanking = exports.Club = exports.BattleLog = void 0;
const BrawlStarsClient_1 = require("./_classes/BrawlStarsClient");
const BrawlStarsApi_1 = require("./_classes/BrawlStarsApi");
const BattleLog_1 = require("./_classes/BattleLog");
const Club_1 = require("./_classes/Club");
const ClubRanking_1 = require("./_classes/ClubRanking");
const Player_1 = require("./_classes/Player");
const PlayerRanking_1 = require("./_classes/PlayerRanking");
class BattleLog extends BattleLog_1.default {
}
exports.BattleLog = BattleLog;
class Club extends Club_1.default {
}
exports.Club = Club;
class ClubRanking extends ClubRanking_1.default {
}
exports.ClubRanking = ClubRanking;
class Player extends Player_1.default {
}
exports.Player = Player;
class PlayerRanking extends PlayerRanking_1.default {
}
exports.PlayerRanking = PlayerRanking;
class Client extends BrawlStarsClient_1.default {
}
exports.Client = Client;
class API extends BrawlStarsApi_1.default {
}
exports.API = API;
exports.default = {
    client: Client,
    api: API,
};
